cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SupportBundleHttpHandler.java

```diff
@@ -0,0 +1,552 @@
+          .create();
+
+  private final NamespaceQueryAdmin namespaceQueryAdmin;
+  private final Store store;
+  private final ApplicationLifecycleService applicationLifecycleService;
+  private final MetricsQueryHelper metricsQueryHelper;
+  private final LogReader logReader;
+  private final String logPattern;
+  private final List<String> serviceList;
+  private final List<String> listOfFileNames;
+  private final CConfiguration cConfig;
+  private final JsonObject serviceJson = new JsonObject();
+
+  @Inject
+  SupportBundleHttpHandler(
+      Store store,
+      NamespaceQueryAdmin namespaceQueryAdmin,
+      ApplicationLifecycleService applicationLifecycleService,
+      MetricsQueryHelper metricsQueryHelper,
+      LogReader logReader,
+      CConfiguration cConfig) {
+    this.store = store;
+    this.namespaceQueryAdmin = namespaceQueryAdmin;
+    this.applicationLifecycleService = applicationLifecycleService;
+    this.metricsQueryHelper = metricsQueryHelper;
+    this.logReader = logReader;
+    this.logPattern =
+        cConfig.get(LoggingConfiguration.LOG_PATTERN, LoggingConfiguration.DEFAULT_LOG_PATTERN);
+    this.serviceList =
+        Arrays.asList(
+            Constants.Service.APP_FABRIC_HTTP,
+            Constants.Service.DATASET_EXECUTOR,
+            Constants.Service.EXPLORE_HTTP_USER_SERVICE,
+            Constants.Service.LOGSAVER,
+            Constants.Service.MESSAGING_SERVICE,
+            Constants.Service.METADATA_SERVICE,
+            Constants.Service.METRICS,
+            Constants.Service.METRICS_PROCESSOR,
+            Constants.Service.RUNTIME,
+            Constants.Service.TRANSACTION,
+            "pipeline");
+    this.listOfFileNames = new ArrayList<>(serviceList);
+    listOfFileNames.addAll(Arrays.asList("applicationFile", "runtimelog", "runtimeinfo"));
+    this.cConfig = cConfig;
+    String homePath = System.getProperty("user.home") + "/support/bundle";
+    cConfig.set(Constants.AppFabric.SUPPORT_BUNDLE_DIR, homePath);
+  }
+
+  /**
+   * Generate the Support Bundle if valid application id, workflow id, and runid are provided.
+   *
+   * @param namespaceId the namespace id
+   * @param appId the app id
+   * @param workflowName the workflow name
+   * @param runId the runid of the workflow
+   * @param needSystemLog whether the customer need system log(appfabric) file or not return the
+   *     uuid of this support bundle
+   * @throws NotFoundException is thrown when the application, workflow, or runid is not found
+   */
+  @POST
+  @Path("/support/bundle")
+  public void createSupportBundle(
+      HttpRequest request,
+      HttpResponder responder,
+      @Nullable @QueryParam("namespace-id") String namespaceId,
+      @Nullable @QueryParam("app-id") String appId,
+      @Nullable @QueryParam("workflow-name") String workflowName,
+      @Nullable @QueryParam("run-id") String runId,
+      @DefaultValue("false") @QueryParam("need-system-log") Boolean needSystemLog)
```

Comment:
You don't need this. Collect everything and then person will select what's needed. We may want to add some filtering later, but by default we should just collect everything